Source/AwsCommonRuntimeKit/io/EventLoopGroup.swift

```diff
@@ -12,24 +12,18 @@ public final class EventLoopGroup {
     public init(threadCount: UInt16 = 0,
                 allocator: Allocator = defaultAllocator,
                 shutDownOptions: ShutDownCallbackOptions? = nil) {
-        var ptr: UnsafePointer<aws_shutdown_callback_options>?
-        if let shutDownOptions = shutDownOptions {
-            let shutDownPtr = UnsafeMutablePointer<ShutDownCallbackOptions>.allocate(capacity: 1)
-            shutDownPtr.initialize(to: shutDownOptions)
-            let options = aws_shutdown_callback_options(shutdown_callback_fn: { (userData) in
-                guard let userdata = userData else {
-                    return
-                }
-                let pointer = userdata.assumingMemoryBound(to: ShutDownCallbackOptions.self)
-                defer { pointer.deinitializeAndDeallocate() }
-                pointer.pointee.shutDownCallback(pointer.pointee.semaphore)
+        let shutDownPtr: UnsafeMutablePointer<ShutDownCallbackOptions>? = fromOptionalPointer(ptr: shutDownOptions)
+        let options = aws_shutdown_callback_options(shutdown_callback_fn: { (userData) in
```

Comment:
I've just realized that we've re-used this exact pattern at least 3 times, we should have a function that produces ShutdownCallbackOptions from a closure.

Reply:
made this!